evm/src/executor/inspector/cheatcodes/mod.rs

```diff
@@ -821,6 +821,15 @@ where
 
         // If the depth is 0, then this is the root call terminating
         if data.journaled_state.depth() == 0 {
+            // See if there's a dangling expectRevert that should've been matched.
+            if self.expected_revert.is_some() {
+                return (
+                    InstructionResult::Revert,
+                    remaining_gas,
+                    "Expected revert was left dangling".encode().into(),
```

Comment:
Nit on phrasing to make it more clear, open to alternatives
```suggestion
                    "A `vm.expectRevert` was left dangling".encode().into(),
```

Reply:
Maybe we could hint at how it behaves? Something like:
```suggestion
                    "A `vm.expectRevert`was left dangling. Make sure that calls you expect to revert are external".encode().into(),
```

wdyt?

Reply:
that sound great to me 👌

Reply:
done in https://github.com/foundry-rs/foundry/pull/4945/commits/3bc04ef3776f93468bf52026cb703fe6992d3285 !